### ISSUE TITLE
Playback-Track: playback offset

### ISSUE THREAD

**Issue by desperateCoder** [hydrogen-music/hydrogen#667]
Hi, its me again, the playback-track-guy.

Thanks to the Playback-Track i was able to get some nice MIDI-files for our professional recording-guy as a base for the rest of recording. Since drum-recording would cost much more we decided to just to use the MIDI for the record.
Now i got a rough mix, where i notice much of potential for the drums, so i took the mix and planned to add some details. Right now i am going crazy getting it tight again, because there are a few silent seconds before the track begins and i cant get this right. 
I tried cutting it with audacitiy and stuff, but the struggle is real. 

TL;DR:
As we got the playback-track, it would be nice to be able to add an offset of X millis / nanos / whatever unit you count in, so i could align the track with my beats. Surely it would be great to be able to add or subtract some seconds, in my case i don't want the first few seconds. There may be cases, where there needs to be some silent seconds before playback begins. 

I hope i could clarify my struggle and i hope there isn't too much effort needed for this.
If you need a tester or have any questions, here i am.

Thank you guys for this whole thing, i love H2! <3

**Comment by theGreatWhiteShark** [comment]
I'm not quite sure if I got your problem right. 

1. You used Hydrogen to create some MIDI files and fed them to an external program to provide the drum track for your recording.
2. Then, after the recording took place, you decided generate all drum tracks using Hydrogen. But the external program you used beforehand introduced an offset and now you struggle to combine the drum track generated by Hydrogen with your recorded material.

Is this what you are talking about?

Is the offset consistent throughout all of your tracks? In this case you should determine its exact length and either introduce some silence of the same lengths at the beginning of all your recorded tracks or introduced the drum track not at 0s but after the offset (depending on the sign of the offset). If your using Linux and the Jack server, you can just pipe the whole output of Hydrogen through a delay.

**Comment by theGreatWhiteShark** [comment]
Okay. I'm neither the one responsible for the playback-track feature nor have I peaked into its implementation. I actually used it for the first time some minutes ago. But in general Hydrogen does not feature much functionality in terms of audio editing (apart from the SampleEditor). In Linux
 people tend to use separate programs for separate tasks and your problem could be e.g. solved with a few lines of bash scripting and the sox command line program https://stackoverflow.com/questions/12562023/trim-audio-files-with-sox-in-milliseconds#12562236. Or with a graphical user interface, as you have already done.

I would recommend you to use the Jack server and some extra beats in the beginning of each song, which do count in. It will keep your recording software (not sure about Audacity, but try the Non-DAW or Ardour) and Hydrogen in synchronization. 

Since you are a beginner, I would recommend you a blog post I wrote some 1.5 half years ago about Linux audio, the major programs, and the different sound servers. https://thegreatwhiteshark.github.io/thegreatwhiteshark.music.io/linux-audio/recording-audio-using-linux-systems/ It might help you to get started.

**Comment by mauser** [comment]
Hi! 

I can understand the intention of the feature request, this is something which also came to my mind when implementing the playback track. But as @theGreatWhiteShark already noted, Hydrogen is not really laid out for this kind of editing and there are still other things lacking about the playback track. So this would be a nice-to-have thing, but i guess a lot of other stuff will be more important..

**Comment by desperateCoder** [comment]
@mauser please see #670

I guess this is the more important stuff, you are talking about

It is totally fine if this feature won't be implemented that quick, or maybe not at all.

Feel free to close this issue, #670 confused me that much, I didn't think of the possibility that the playback speed is wrong. Leaving one whole note empty at the beginning of the right-speed-track resolved my issue.

**Comment by elpescado** [comment]
We already have sample editor - maybe we could enable it on playback track?
